src/sentry/auth/providers/google/__init__.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import absolute_import
+
+from django.conf import settings
+
+from sentry import auth, options
+
+from .provider import GoogleOAuth2Provider
+
+auth.register('google', GoogleOAuth2Provider)
+
+options.register(
+    'auth-google.client-id',
+    default=getattr(settings, 'GOOGLE_CLIENT_ID', None),
```

Comment:
Before we merge this, this isn't the same behavior. The default here should be empty, but if we wanna prefill data into options, we should do that inside of `runner/initializer.py` so it gets shoved actually into `SENTRY_OPTIONS` dict.

Reply:
To be explicit, I'd recommend in the `apply_legacy_settings` function in there.

Reply:
Is there a better pattern we could provide that explains where to obtain a default from? You could imagine an upstream solution wanted to do this, so seems like something maybe we should see if we can't solve.

Reply:
I wouldn't say this is only for porting settings. It is - but to some degree it might be more along the lines of "ive defined the settings elsewhere" or "ive defined defaults to fall back to".

Reply:
Well, then the only thing they need to do is provide their own hook to shove into `SENTRY_OPTIONS` explicitly. We just don't have any easy hooks or docs to do this.

But if you wanted to deal with something here, the better option would be to do like:

```
SENTRY_OPTIONS['auth-google.client-id'] = getattr(settings, 'GOOGLE_CLIENT_ID', None)
```

Inside this file. As a way to fill in from somewhere else or to fit your example.

The reason for this is there's slightly different behavior with what I'm suggesting vs the default. default is checked _after_ an attempt to query a value from the database, falling back to a default. But if it's set into SENTRY_OPTIONS explicitly, it doesn't attempt to query for it.